internal/interpreter/api.rs

```diff
@@ -626,6 +626,22 @@ impl ComponentDefinition {
         })
     }
 
+    /// Instantiate the component using an existing window.
+    #[doc(hidden)]
+    pub fn embed_into_existing_window(
```

Comment:
We need a nice public API for that.
Would be nice to combine with the other function.

Reply:
As mentioned before in chat, I'd rather this method is not introduced as it breaks the laziness of the window adapter creation. There's nothing inherent about embedding requiring the adapter. I think, instead, an embedded component should reach out to its parent component when looking for the adapter.

Reply:
I realize that the laziness issue is minor. But I'm more concerned that we're creating the wrong public factory API here that has a Window parameter not because it has any meaning to our users but because it merely simplifies an implementation detail on our side.

Worse: if users don't pass the Window parameter correctly but accidentally call the create function without window it works but gives funny effects (duplicate adapter).

Reply:
One option is to give the factory an opaque token argument and the component returned must be built with that token. That's how we could smuggle the window through.

I'd much rather prefer the direction you hinted at though: Adding the necessary internal APIs to component vtable to make this work without bothering users. From my changes with the lazy adapters I think that means `window_adapter()` that returns the adapter and creates if necessary, and `maybe_window_adapter()` that returns one of exists and None of not (needed by Drop/destructor).

Reply:
the problem is that the "lazy" window creation might not be lazy enough.  The init call can query the window, or the user querying a property before returning from the factory.
Maybe that means we should decouple the things that need the window from the window. Such as the scale factor, the dark theme and so on. (ie, these property could be accessed through another mean that the Window)
But that might be out of scope for this patch now.

Reply:
I agree, those properties could be moved out of the adapter. Perhaps stored in the compo, with accessor. Embedded components could use a two-way binding.